pkg/catalog/catalog.go

```diff
@@ -5,7 +5,9 @@ import (
 	"strings"
 	"sync"
 
-	"github.com/hashicorp/go-version"
+	"github.com/Masterminds/semver/v3"
+	"github.com/fluxcd/pkg/version"
+
```

Comment:
I'm OOL, was there actually anything wrong with the previous version parser? We use it in eksctl and it works well :shrug:

Reply:
This is an attempt at unifying the work we do as a company. /cc @bigkevmcd 
We can skip it if you think that it should not be included ATM.

Reply:
nah thats fine